main.go

```diff
@@ -419,38 +254,26 @@ func main() {
 	if err := cfg.validate(); err != nil {
 		fail("Input error: %s", err)
 	}
+	// Select authentication source
+	authSources, err := parseAuthSources(cfg.BitriseAppleConnection)
+	if err != nil {
+		fail("Input error: unexpected value for Bitrise Apple Developer Connection (%s)", cfg.BitriseAppleConnection)
+	}
 
 	//
-	// Fastlane session
-	fastlaneSession := ""
-	buildURL, buildAPIToken := os.Getenv("BITRISE_BUILD_URL"), os.Getenv("BITRISE_BUILD_API_TOKEN")
-	if buildURL != "" && buildAPIToken != "" {
-		var provider devportalservice.AppleDeveloperConnectionProvider
-		provider = devportalservice.NewBitriseClient(http.DefaultClient)
-
-		conn, err := provider.GetAppleDeveloperConnection(buildURL, buildAPIToken)
-		if err != nil {
-			handleSessionDataError(err)
-		}
-
-		if conn != nil && conn.SessionConnection != nil {
-			fmt.Println()
-			log.Infof("Connected session-based Apple Developer Portal Account found")
-
-			sessionConn := conn.SessionConnection
-
-			if sessionConn.AppleID != cfg.ItunesConnectUser {
-				log.Warnf("Connected Apple Developer and App Store login account missmatch")
-			} else if expiry := sessionConn.Expiry(); expiry != nil && sessionConn.Expired() {
-				log.Warnf("TFA session expired on %s", expiry.String())
-			} else if session, err := sessionConn.FastlaneLoginSession(); err != nil {
-				handleSessionDataError(err)
-			} else {
-				fastlaneSession = session
-			}
-		}
-	} else {
-		log.Warnf("Step is not running on bitrise.io: BITRISE_BUILD_URL and BITRISE_BUILD_API_TOKEN envs are not set")
+	// Fetch Apple authenication source
+	authConfig, err := FetchAppleAuthData(authSources, AppleAuthInputs{
+		Username:            cfg.ItunesConnectUser,
+		Password:            string(cfg.Password),
+		AppSpecificPassword: string(cfg.AppPassword),
+		APIIssuer:           cfg.APIIssuer,
+		APIKeyPath:          cfg.APIKeyPath,
+	})
+	if err != nil {
+		fail("Could not configure App Store Connect authentication: %v", err)
```

Comment:
Could not configure App Service authentication 
(in case of apple id auth, spaceship uses the legacy API: https://github.com/fastlane/fastlane/blob/master/spaceship/docs/AppStoreConnect.md#usage
not the App Store Connect API)